test/setup/versions.tf

```diff
@@ -19,10 +19,10 @@ terraform {
 
   required_providers {
     google = {
-      version = ">= 3.49, < 5.0"
+      version = ">= 3.49, < 5.3"
```

Comment:
what is the reason for `version = ">= 3.49, < 5.3"` instead of `version = ">= 3.49, <= 5.2"` ?

Reply:
As this is just for CI test setup, we can drop the max constraint entirely.